api-server/src/server/utils/donation.js

```diff
@@ -171,3 +172,95 @@ export async function updateUser(body, app) {
       type: 'UnsupportedWebhookType'
     };
 }
+
+export function hasDonatedToday(donations) {
+  if (donations.length > 1) {
+    const milliSecondsInDay = 86400000;
+    const milliSecondsNow = new Date().getTime();
+    return donations.some(donation => {
+      return (
+        milliSecondsNow - new Date(donation.startDate['_date']).getTime() <
+        milliSecondsInDay
+      );
+    });
+  }
+  return false;
+}
+
+export async function createStripeCardDonation(req, res, stripe, app) {
+  const {
+    body: {
+      token: { id: tokenId },
+      amount,
+      duration
+    },
+    user: { name, id: userId, email },
+    user
+  } = req;
+
+  if (!tokenId || !amount || !duration || !name || !userId || !email) {
+    throw {
+      message: 'Request is not valid',
+      type: 'InvalidRequest'
+    };
+  }
+
+  // check request validity
+  const { Donation } = app.models;
+
+  Donation.find({ where: { userId } }, (err, donations) => {
+    if (err) throw Error(err);
+
+    // check if a user has already made multiple donations
+    if (donations.length > 10)
```

Comment:
Do we never have a case where a Camper legitimately donates `> 10` times?

Reply:
Great question. No this is a check to make sure we stop bots and allow legit users. 
I don't know if we have a user with more than 5 legit donations throughout the years.
@raisedadead, would you like to block users with more than a certain donations and users who have recently donated, or should I just block any user who has ever donated?

My logic is that we would be encouraging donors to make additional donations in the future, but it is not the case yet.

Reply:
What about back when we allowed one time donations? I can see someone getting to 10 if we were to reinstate that.

Reply:
Great point. To my knowledge, one time donations were not being saved in the past, but we can always revisit this if we need to make changes to our donation structure.

Reply:
For simplicity, we will just check for isDonating for this iteration.